internal/provisioner/controller/gateway.go

```diff
@@ -272,9 +281,15 @@ func (r *gatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 			}
 
 			// Deployment replicas
-			if gatewayClassParams.Spec.Envoy.WorkloadType == contour_api_v1alpha1.WorkloadTypeDeployment &&
-				gatewayClassParams.Spec.Envoy.Replicas > 0 {
-				contourModel.Spec.EnvoyReplicas = gatewayClassParams.Spec.Envoy.Replicas
+			if gatewayClassParams.Spec.Envoy.WorkloadType == contour_api_v1alpha1.WorkloadTypeDeployment {
+				if gatewayClassParams.Spec.Envoy.Deployment != nil && gatewayClassParams.Spec.Envoy.Deployment.Replicas > 0 {
+					contourModel.Spec.EnvoyReplicas = gatewayClassParams.Spec.Envoy.Deployment.Replicas
+
+				} else if gatewayClassParams.Spec.Envoy.Replicas > 0 { //nolint: staticcheck
```

Comment:
what lint error is this and the below ignoring?

also seems like it could be easier to just set `contourModel.Spec.EnvoyReplicas` to `gatewayClassParams.Spec.Envoy.Replicas` if specified and then override it to `gatewayClassParams.Spec.Envoy.Deployment.Replicas` if that is specified, would prevent us from having as many `if` blocks

Reply:
SA1019: gatewayClassParams.Spec.Contour.Replicas is deprecated: Use `DeploymentSettings.Replicas` instead. (staticcheck)